describe_log_dirs_request.go

```diff
@@ -21,34 +21,68 @@ type DescribeLogDirsRequestTopic struct {
 }
 
 func (r *DescribeLogDirsRequest) encode(pe packetEncoder) error {
-	length := len(r.DescribeTopics)
-	if length == 0 {
-		// In order to query all topics we must send null
-		length = -1
-	}
-
-	if err := pe.putArrayLength(length); err != nil {
-		return err
-	}
+	isFlexible := r.Version >= 2
 
-	for _, d := range r.DescribeTopics {
-		if err := pe.putString(d.Topic); err != nil {
+	if isFlexible {
+		if len(r.DescribeTopics) == 0 {
+			pe.putUVarint(0)
+		} else {
+			pe.putCompactArrayLength(len(r.DescribeTopics))
+		}
+	} else {
+		length := len(r.DescribeTopics)
+		if length == 0 {
+			// In order to query all topics we must send null
+			length = -1
+		}
+		if err := pe.putArrayLength(length); err != nil {
 			return err
 		}
+	}
 
-		if err := pe.putInt32Array(d.PartitionIDs); err != nil {
-			return err
+	for _, d := range r.DescribeTopics {
+		if isFlexible {
+			if err := pe.putCompactString(d.Topic); err != nil {
+				return err
+			}
+
+			if err := pe.putCompactInt32Array(d.PartitionIDs); err != nil {
+				return err
+			}
+			if isFlexible {
```

Comment:
This test is already known to be true, because we’re in an `if isFlexible` conditional.